pkg/steps/source.go

```diff
@@ -733,3 +710,30 @@ func istObjectReference(ctx context.Context, client ctrlruntimeclient.Client, re
 	}
 	return corev1.ObjectReference{Kind: "DockerImage", Name: fmt.Sprintf("%s@%s", repo, ist.Image.Name)}, nil
 }
+
+func cloneRefsOptions(jobSpec *api.JobSpec, cloneAuthConfig *CloneAuthConfig) clonerefs.Options {
```

Comment:
Is there not something upstream to do this for us?

Reply:
For this particular bit here no, for what the pod step does kind-of. The upstream code requires a prowjob which we don't have, but we have the required attributes for said prowjob. I just constructed one for the pod step, WDYT?

Reply:
Ah, gotcha - this approach sounds good